pkg/clusterd/disk.go

```diff
@@ -66,6 +66,24 @@ func DiscoverDevices(executor exec.Executor) ([]*sys.LocalDisk, error) {
 			// go on without udev info
 			logger.Warningf("failed to get udev info for device %s: %+v", d, err)
 		}
+
+		// Test if device has child, if so we skip it and only consider the partitions
+		// which will come in later iterations of the loop
+		// We only test if the type is 'disk', this is a property reported by lsblk
+		// and means it's a parent block device
+		if disk.Type == sys.DiskType {
+			deviceChild, err := sys.ListDevicesChild(executor, d)
```

Comment:
This can return error and still have devices. In general, I would expect that if there were an error, the normal output wouldn't be valid (I would expect a `continue` statement in the err checking block). Is it true in this case that `deviceChild` can still be valid in the error case? Is there **any** case where `deviceChild` can be invalid?

Reply:
If the check for a child fails, we can assume they are none, thus we shouldn't `continue` and still report that device. Makes sense?

Reply:
I see. We still need to reach the `disks = append(disks, disk)` line.